README.md

```diff
@@ -1 +1,122 @@
+    ::Type{CT}=Tuple{} # ignore this for now
+) where {CT}
+```
+Let's break it down:
+* `lexer` controls how we find newlines in the ingested chunks of data. Newlines serve as record delimiters and knowing their positions allows us to split work safely among multiple workers, which are spawned internally. `Lexer`s are defined in the `NewlineLexers.jl` package.
+* `parsing_ctx` controls how we parse the data. It allows the user to dispatch on custom `populate_result_buffer!` overload and to forward configurations to it. `populate_result_buffer!` is where we take the records identified by the `lexer` and parse them into `result_buffers`.
+* `consume_ctx` controls how the parsed results are consumed (e.g. inserted into a database, appended to a `DataFrame`...). `consume_ctx` allows the user to dispatch on their `consume!` method and hold any state necessary for consumption. This happens immediately after `populate_result_buffer!`.
+* `chunking_ctx` controls how the work on individual chunks of data is scheduled. It contains buffers for input bytes and found newlines. Through this struct the user controls the size of the chunks and the number of spawned tasks that carry out the parsing and consuming. If there is enough data in the input, a secondary `chunking_ctx` is created internally to facilitate the double-buffering described above.
+* `result_buffers` controls in which format the results are stored. These result buffers hold results from `populate_result_buffer!` and are passed to `consume!`. This allows the user to have multiple result formats for the with `parsing_ctx` e.g. row oriented vs column oriented buffers.
+
+See the docstring of `populate_result_buffer!` and `consume!` for more information about how to integrate with them.
+
+## Example: Examining the results of the Lexer
+
+Since `ChunkedBase.jl` handles newline detection automatically, a very simple chunked processor could just show the newline positions to the user. Of course, this is not very useful apart from demonstrating how to use this package.
+
+```julia
+using ChunkedBase
+using NewlineLexers: Lexer
+
+# Our chunked processor doesn't require any settings nor does it need to maintain
+# any additional state we'll define our `ParsingContext` and `ConsumeContext` only
+# for dispatch reasons.
+struct ParsingContext <: AbstractParsingContext end
+struct ConsumeContext <: AbstractConsumeContext end
+# Our buffer will hold the newline positions in the current chunk
+struct ResultBuffer <: AbstractResultBuffer
+    newlines_in_chunk::Vector{Int}
+end
+
+# Our overload copies `newlines_segment` to our result buffer
+function ChunkedBase.populate_result_buffer!(
+    result_buf::ResultBuffer,
+    newlines_segment::AbstractVector{Int32},
+    ::ParsingContext,
+    ::Vector{UInt8},
+    ::Union{Nothing,Vector{UInt8}}=nothing,
+    ::Type=Tuple{}
+)
+    resize!(result_buf.newlines_in_chunk, length(newlines_segment))
+    result_buf.newlines_in_chunk .= copy(newlines_segment)
```

Comment:
super nitpicky, since this is just an example, but I think you don't need the copy() here, since `.=` is already going to copy the data from the RHS into the LHS :)